### ISSUE TITLE
http://knilxof.org:4242/ping shows incorrect server info when multiple foxboxes are registered in the same network.

### ISSUE THREAD

**Issue by nojunpark** [fxbox/foxbox#245]
It seems that when there are more than one foxbox app running, and one of them is terminated, its info does not disappear completely - apparently the other foxbox app info takes the place instead, duplicating the information, shown as below.   In below instance, there was only one gmarty.knilxof.org running, and another foxbox process was ran and terminated repeatedly on 10.252.25.42.  (it shows as gmarty ip below)

[
{
    "public_ip": "63.245.221.32",
    "local_ip": "10.252.25.38",
    "tunnel_url": "gmarty.knilxof.org",
    "timestamp": ​1458592404
},
{
    "public_ip": "63.245.221.32",
    "local_ip": "10.252.25.42",
    "tunnel_url": "gmarty.knilxof.org",
    "timestamp": ​1458592436
},
{

```
"public_ip": "63.245.221.32",
"local_ip": "10.252.120.212",
"tunnel_url": "gmarty.knilxof.org",
"timestamp": ​1458592449
```

},
    {
        "public_ip": "63.245.221.32",
        "local_ip": "10.252.25.33",
        "tunnel_url": "azasypkin.knilxof.org",
        "timestamp": ​1458592453
    }
]

**Comment by nojunpark** [comment]
ni? @fabricedesre  for info

**Comment by nojunpark** [comment]
Related issue is that it takes about 15 minutes or so for an entry to be removed, causing duplicate entries in http://knilxof.org:4242/ping as follows:
[

{

```
"public_ip": "135.23.252.205",
"local_ip": "192.168.1.142",
"tunnel_url": "integrationtestonly.knilxof.org",
"timestamp": ​1459308922
```

},

```
{
    "public_ip": "135.23.252.205",
    "local_ip": "192.168.1.131",
    "tunnel_url": "integrationtestonly.knilxof.org",
    "timestamp": ​1459309454
}
```

]

**Comment by fabricedesre** [comment]
If the local_ip is different it's not a duplicate. Or are you also seeing multiple records with the same public and local ips?

**Comment by nojunpark** [comment]
no, the problem is minor in this case. When above ping server was queried, the top entry(1.142) was already terminated, but the ping list wasn't updated instantly.  (it takes about 15 min for the entry to disappear)  So the user would think both entries are active, which is not the case.    In my test script, I get around this by picking the one with the latest timestamp.

**Comment by fabricedesre** [comment]
The eviction timeout for stalled entries is 10 minutes (see https://github.com/fxbox/registration_server/blob/master/src/eviction.rs#L9). This may be too long indeed (clients are expected to update their state every minute).
Is that a problem when running tests?

**Comment by nojunpark** [comment]
not anymore now I have the timestamp check in place. had I known this earlier, I would have saved time debugging the test though :) I'm thinking 1 or 2 min would be nice, if possible.

**Comment by samgiles** [comment]
This behaviour shouldn't exist at master, because there is also a unique
client I'd, based on the self signed cert each box has.
